### PR TITLE
adding node kafka rest client

### PR DESCRIPTION
**Pull request by fx19880617**
If proxyPort is not set, then it won't initialized or called.

Test results:

=============================================================================
-----------------------------------------|----------|----------|----------|----------|----------------|
File                                     |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
-----------------------------------------|----------|----------|----------|----------|----------------|
 kafka-logger-xiangfu/                   |    75.86 |    63.74 |     87.5 |    75.86 |                |
  index.js                               |    75.86 |    63.74 |     87.5 |    75.86 |... 182,207,216 |
 kafka-logger-xiangfu/kafka-rest-client/ |    96.27 |    81.48 |      100 |    96.27 |                |
  kafka_producer.js                      |     91.3 |    65.38 |      100 |     91.3 |... 47,65,78,79 |
  kafka_rest_client.js                   |      100 |    96.43 |      100 |      100 |                |
-----------------------------------------|----------|----------|----------|----------|----------------|
All files                                |    87.73 |    70.34 |     97.3 |    87.73 |                |
-----------------------------------------|----------|----------|----------|----------|----------------|

1..56
tests 56
pass  56
ok

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -48,12 +49,30 @@ function KafkaLogger(options) {
         }
     }
 
+
+    function onKafkaRectClientConnect(err) {
```
Comment (Raynos) [inline index.js]: nitpick: s/Rect/Rest/

File: index.js [hunk 2]
```diff
@@ -48,12 +49,30 @@ function KafkaLogger(options) {
...
+        if (!err) {
+            if (self.logger) {
+                self.logger.info('KafkaRestClient connected to kafka');
+            }
+            self.kafkaRectClientConnected = true;
+        } else {
+            if (self.logger) {
+                self.logger.warn('KafkaRestClient could not connect to kafka');
```
Comment (Raynos) [inline index.js]: This should be `self.logger.error()`

Comment (Raynos) [inline index.js]: Never mind; let's keep it warn.

File: index.js [hunk 3]
```diff
@@ -48,12 +49,30 @@ function KafkaLogger(options) {
...
+
     options = options || {};
 
     Transport.call(this, options);
     this.topic = options.topic || 'unknown';
     this.leafHost = options.leafHost || 'localhost';
     this.leafPort = options.leafPort || 9093;
+    this.proxyHost = options.proxyHost || 'localhost';
+    if ('proxyPort' in options && options.proxyPort) {
+      this.proxyPort = options.proxyPort;
```
Comment (Raynos) [inline index.js]: 4 spaces please.

File: index.js [hunk 4]
```diff
@@ -77,8 +96,20 @@ function KafkaLogger(options) {
     this.isDisabled = options.isDisabled || null;
 
     this.connected = true;
+    this.kafkaRectClientConnected = false;
```
Comment (Raynos) [inline index.js]: s/Rect/Rest/

File: index.js [hunk 5]
```diff
@@ -77,8 +96,20 @@ function KafkaLogger(options) {
...
+    if (!this.kafkaRestClient) {
+        if (this.proxyPort) {
+            this.kafkaRestClient = new KafkaRestClient({
+                proxyHost: this.proxyHost,
+                proxyPort: this.proxyPort
+            });
+            this.kafkaRestClient.connect(onKafkaRectClientConnect);
+        }
+    } else {
+        this.kafkaRectClientConnected = true;
```
Comment (Raynos) [inline index.js]: this looks like a dead branch to me.

File: package.json [hunk 6]
```diff
@@ -7,8 +7,9 @@
     "test": "test"
   },
   "scripts": {
-    "cover": "istanbul cover --report none --print detail _mocha -- test --reporter tap",
+    "cover": "istanbul cover test/index.js  --report none --print detail _mocha -- test --reporter tap",
```
Comment (Raynos) [inline package.json]: lets get rid of the _mocah stuff.

File: package.json [hunk 7]
```diff
@@ -36,13 +37,19 @@
...
     "winston-uber": "^1.0.0",
-    "xtend": "^4.0.0"
+    "xtend": "^4.0.0",
+    "error": "^7.0.1",
+    "lb_pool": "^1.2.0"
   },
   "devDependencies": {
     "istanbul": "~0.1.46",
     "tape": "^3.0.3",
+    "mocha": "*",
```
Comment (Raynos) [inline package.json]: Lets not add mocha

File: package.json [hunk 8]
```diff
@@ -36,13 +37,19 @@
...
     "uber-licence": "^1.5.1",
     "uber-nodesol": "^0.3.12-yosemite",
-    "uuid": "~1.4.1"
+    "uuid": "~1.4.1",
+    "itape": "^1.5.0",
+    "rewire": "*",
```
Comment (Raynos) [inline package.json]: do not use *

**Comment by Raynos** [comment]
Looks good enough; Fix the minor comments and lets get this shipped.

File: index.js (hunk 5 above)
Comment (fx19880617) [inline index.js]: Only if someone init the client and pass it in.

File: package.json (hunk 6 above)
Comment (fx19880617) [inline package.json]: Removed

File: package.json (hunk 7 above)
Comment (fx19880617) [inline package.json]: Done

File: package.json (hunk 8 above)
Comment (fx19880617) [inline package.json]: done

File: index.js (hunk 1 above)
Comment (fx19880617) [inline index.js]: done for all

File: index.js (hunk 3 above)
Comment (fx19880617) [inline index.js]: done

File: index.js (hunk 4 above)
Comment (fx19880617) [inline index.js]: done

**Comment by Raynos** [comment]
Looks good; feel free to merge.

**Pull request merged**